priority_queue.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+
+class PriorityQueue(object):
```

Comment:
You've gotten the right idea that the binary heap is the basis for the priority queue, but you've completely re-written the heap code instead of composing a heap into this structure.  Consider this:

```python
from binary_heap import BinaryHeap

class PriorityQueue(object):

    def __init__(self):
        self.heap = BinaryHeap()
```

What happens to the rest of your code if you start that way?  What prevents you from doing so?  Does what you see as a stumbling block really block you?  Or is there a way around it?